Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -67,17 +127,30 @@ extension HTTPClient {
         ///     - length: Body size. Request validation will be failed with `HTTPClientErrors.contentLengthMissing` if nil,
         ///               unless `Transfer-Encoding: chunked` header is set.
         ///     - stream: Body chunk provider.
+        @available(*, deprecated, message: "StreamWriter is deprecated, please use StreamWriter2 instead")
         public static func stream(length: Int? = nil, _ stream: @escaping (StreamWriter) -> EventLoopFuture<Void>) -> Body {
             return Body(length: length, stream: stream)
         }
 
+        /// Create and stream body using `StreamWriter`.
+        ///
+        /// - parameters:
+        ///     - length: Body size. Request validation will be failed with `HTTPClientErrors.contentLengthMissing` if nil,
+        ///               unless `Transfer-Encoding: chunked` header is set.
+        ///     - stream: Body chunk provider.
+        public static func stream2(length: Int? = nil, _ stream: @escaping (StreamWriter2) -> Void) -> Body {
```

Comment:
I we add method with the same name but different type we will break compilation, because some calls could become ambiguous 😢

Reply:
Given that there is discussion about possibly breaking API in the nearish future _anyway_, we may find we want to delay this until then to resolve this problem.